tests/unittest_brain.py

```diff
@@ -1678,7 +1678,310 @@ def make_new_type(t):
         """
         )
         with self.assertRaises(UseInferenceDefault):
-            astroid.brain.brain_typing.infer_typing_typevar_or_newtype(node.value)
+            astroid.brain.brain_typing.infer_typing_newtype(node.value)
+
+    def test_typing_newtype_attrs(self) -> None:
+        ast_nodes = builder.extract_node(
+            """
+        from typing import NewType
+        import decimal
+        from decimal import Decimal
+
+        NewType("Foo", str) #@
+        NewType("Bar", "int") #@
+        NewType("Baz", Decimal) #@
+        NewType("Qux", decimal.Decimal) #@
+        """
+        )
+        assert isinstance(ast_nodes, list)
+
+        # Base type given by reference
+        foo_node = ast_nodes[0]
+
+        # Should be unambiguous
+        foo_inferred_all = list(foo_node.infer())
+        assert len(foo_inferred_all) == 1
+
+        foo_inferred = foo_inferred_all[0]
+        self.assertIsInstance(foo_inferred, astroid.ClassDef)
+
+        # Check base type method is inferred by accessing one of its methods
+        foo_base_class_method = foo_inferred.getattr("endswith")[0]
+        self.assertIsInstance(foo_base_class_method, astroid.FunctionDef)
+        self.assertEqual("builtins.str.endswith", foo_base_class_method.qname())
```

Comment:
Feel free to ignore this, but I would be okay with changing this to use `assert` instead of `self.assert...`. But that's quite a bit of work perhaps, so feel free to leave as is.

Reply:
Done